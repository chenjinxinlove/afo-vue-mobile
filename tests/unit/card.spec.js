import Vue from 'vue'
import Card from '../../src/card/index'
import Demo from '../../src/card/demo'
import {shallowMount, mount} from '@vue/test-utils'

describe('Card.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Card)
    expect(Vue.component(Card.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 是否能正确渲染
  it('should render correct contents', () => {
    const wrapper = shallowMount(Card, {
      propsData: {
        desc: '描述信息'
      }
    })
    expect(wrapper.find('.afo-card--label').text()).toBe('描述信息')
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Card, {
      propsData: {
        title: '标题',
        desc: '描述信息'
      },
      slots: {
        header: '<button>按钮</button>',
        footer: '<button>按钮</button>'
      }
    })
    expect(wrapper.find('button').text()).toBe('按钮')
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo).html()).toMatchSnapshot()
  })
})
