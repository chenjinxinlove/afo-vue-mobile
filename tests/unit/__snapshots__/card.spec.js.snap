// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue demo snapshot 1`] = `
<div>
  <div class="afo-card">
    <div class="afo-card-header afo-card--cell">
      <div class="afo-card--title"><span>标题</span>
        <div class="afo-card--label">描述信息</div>
      </div>
    </div>
    <div class="afo-card-content afo-card--cell">
      <div>内容</div>
    </div>
    <!---->
  </div>
  <div class="afo-card">
    <div class="afo-card-header afo-card--cell">
      <div class="afo-card--title"><span>标题</span>
        <div class="afo-card--label">描述信息</div>
      </div>
    </div>
    <div class="afo-card-content afo-card--cell">
      <div>内容</div>
    </div>
    <div class="afp-card-footer afo-card--cell afo-hairline--top">
      <div>
        <afo-button size="small">按钮</afo-button>
        <afo-button size="small" type="danger">按钮</afo-button>
      </div>
    </div>
  </div>
</div>
`;
